src/disco/containers/InstallButton.js

```diff
@@ -29,12 +32,53 @@ export class InstallButton extends React.Component {
     downloadProgress: 0,
   }
 
+  componentDidMount() {
+    const { dispatch, guid, installUrl, slug } = this.props;
+    this.addonManager = new AddonManager(guid, installUrl, this.statusChanged);
+    this.addonManager.getAddon().then(() => {
```

Comment:
I broke this onto another line. Apparently `.then(() => {}).catch(() => {})` behaves differently than `.then(() => {}, () => {})` because there is a test failure with the former.